dashboard/src/pages/Hardware/HardwareTable.tsx

```diff
@@ -0,0 +1,244 @@
+      to: '/hardware/$hardwareId',
+      params: { hardwareId: row.original.hardwareName },
+      search: previousSearch => previousSearch,
+    };
+  }, []);
+
+  const data = useMemo(() => {
+    return treeTableRows;
+  }, [treeTableRows]);
+
+  const table = useReactTable({
+    data,
+    columns,
+    onSortingChange: setSorting,
+    onColumnFiltersChange: setColumnFilters,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    onPaginationChange: setPagination,
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    state: {
+      sorting,
+      columnFilters,
+      pagination,
+    },
+  });
+
+  const groupHeaders = table.getHeaderGroups()[0].headers;
+  const tableHeaders = useMemo((): JSX.Element[] => {
+    return groupHeaders.map(header => {
+      return (
+        <TableHead key={header.id}>
+          {header.isPlaceholder
+            ? null
+            : flexRender(header.column.columnDef.header, header.getContext())}
+        </TableHead>
+      );
+    });
+    // TODO: remove exhaustive-deps and change memo (all tables)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [groupHeaders, sorting]);
+
+  const modelRows = table.getRowModel().rows;
+  const tableBody = useMemo((): JSX.Element[] | JSX.Element => {
+    return modelRows?.length ? (
+      modelRows.map(row => (
+        <TableRow key={row.id}>
+          {row.getVisibleCells().map(cell => {
+            return (
+              <TableCellWithLink
+                key={cell.id}
+                linkClassName="w-full inline-block h-full"
+                linkProps={getLinkProps(row)}
+              >
+                {flexRender(cell.column.columnDef.cell, cell.getContext())}
+              </TableCellWithLink>
+            );
+          })}
+        </TableRow>
+      ))
+    ) : (
+      <TableRow>
+        <TableCell colSpan={columns.length} className="h-24 text-center">
+          <FormattedMessage id="global.noResults" />
+        </TableCell>
+      </TableRow>
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [getLinkProps, modelRows, sorting]);
+
+  const MemoizedInputTime = memo(InputTime);
+
+  return (
+    <div className="flex flex-col gap-6 pb-4">
+      <div className="flex items-center justify-between gap-4">
+        <span className="text-left text-sm text-dimGray">
+          <FormattedMessage
+            id="global.projectUnderDevelopment"
+            values={formattedBreakLineValue}
+          />
+        </span>
+        <div className="flex items-center justify-between gap-10">
+          <MemoizedInputTime />
+          <PaginationInfo table={table} data={data} intlLabel="global.trees" />
+        </div>
+      </div>
+      <BaseTable headerComponents={tableHeaders}>
+        <TableBody>{tableBody}</TableBody>
+      </BaseTable>
+      <PaginationInfo table={table} data={data} intlLabel="global.trees" />
```

Comment:
ditto about Trees/Hardwares
```suggestion
      <PaginationInfo table={table} data={data} intlLabel="global.hardwares" />
```